### DISCUSSION TITLE
Connecting to Multiple Printers, not able to connect to the second instance from within Fluidd

### DISCUSSION THREAD

**Discussion by HereticDezign** [fluidd-core/fluidd#1265] [Q&A] [answered]
I posted this on the Klipper Discourse page and was advised by a supercilious user to post my issue here,

I am attempting to get fluidd to connect to multiple printers on a single Rpi3B+ and failing miserably, I can only connect using the i.p. address for some reason using any one the other combinations fails.

I can connect to both MCU’s in Mainsail and check both MCU’s via KIAUH with no problems.

I want to run fluidd because I like its look and feel better than mainsail.

I can connect to an instance using 192.168.0.113 but as soon as i try to add a second printer 192.168.0.113/7126 from within Fluidd it accepts the i.p. and port but it borks everything when I reload and wont reconnect to the first instance until I clear my browser cache.

I then did a fresh install (SD card WIPED) and re-installed everything (Armbian, Klipper & Moonraker via KIAUH) this time including mainsail as the first GUI

When I enter into mainsail it allows me to add both printers in the following manner

192.168.0.113/7125 (printer 1)
192.168.0.113/7126 (printer 2)

That all connects and works fine and i add the IP address to the Cors Domains in each printers moonraker.cfg contained in each printers data folder.

That yields mainsail still connecting fine, fluidd still connects to the first instance but still borks everything again as soon as I try to add a second printer from within fluidd itself.

With Mainsail installed I added fluidd under port:81 via KIAUH and can connect to the first instance of Fluidd via : 192.168.0.113:81 but still cant add a second printer.

Anyone have any insights as to what is going on

Thank you in advance for any help provided.

[klippy.log](https://github.com/fluidd-core/fluidd/files/13615693/klippy.log)

**Comment by pedrolamas** [comment]
Hi @HereticDezig, can you tell me what address are you using in Fluidd when you add a new instance?

Note that it should be with the format `host:port` (a colon separating both, not slash)

Please take a screenshot of the right sidebar open showing the addresses that you added to Fluidd and post it here.

**Reply by HereticDezign** [reply]
As i already listed

> : With Mainsail installed I added fluidd under port:81 via KIAUH and can connect to the first instance of Fluidd via : 192.168.0.113:81 but still cant add a second printer.

That is exactly as you noted.
 
Technically to a point i cant show you what Fluidd added because  once i do it and save, it refreshes then it stops everything dead in its tracks see screenshot series below 

This shows Mainsail seeing both the printers which proves that mainsail/moonraker is both connected and configured correctly too

![mainsail 1](https://github.com/fluidd-core/fluidd/assets/152097995/7cc38537-057a-41f0-a6f8-ff4585f65547)

And connected to Printer 1 in Mainsail

![mainsail  printer 1](https://github.com/fluidd-core/fluidd/assets/152097995/228def08-e6c2-4032-8c46-0e542050c9c9)

Connected to Printer 2 in Mainsail

![mainsail  printer 2](https://github.com/fluidd-core/fluidd/assets/152097995/68930bbf-7acc-4869-950e-1f7ed25482e7)

Connected To Printer 1 in Fluidd

![fluidd printer 1](https://github.com/fluidd-core/fluidd/assets/152097995/72f23c6f-4d01-4c21-ace3-6730d086880c)

Side Bar Open in Printer 1 in Fluidd

![fluidd printer 1 side bar open](https://github.com/fluidd-core/fluidd/assets/152097995/533383b9-cebd-466a-b904-f3a7d05524f6)

Attempting to add a second printer in Fluidd, it shows that Fluidd is happy with the address that has been input

![fluidd printer 1 adding a second printer](https://github.com/fluidd-core/fluidd/assets/152097995/65aaa401-1f81-435d-918c-d61b20fff1f9)

The result of hitting "save" which says its a moonraker connection issue, which makes no logical sense as both Mainsail and Fluidd use the same "printer1_data, printer2_data" moonraker config folders/files as Mainsail can connect to Moonraker ok but Fluidd cant, so logic dictates its a Fluidd issue. and i have cross referenced the moonraker config files with the fluidd documentation on multiple instances and my config parameters are what fluidd requires 

![fluidd printer 1 adding a second printer error message](https://github.com/fluidd-core/fluidd/assets/152097995/61709727-fd72-4b01-a5c1-f89ca534e8fb)

**Reply by pedrolamas** [reply]
> Attempting to add a second printer in Fluidd, it shows that Fluidd is happy with the address that has been input
> 
> ![fluidd printer 1 adding a second printer](https://private-user-images.githubusercontent.com/152097995/289357654-65aaa401-1f81-435d-918c-d61b20fff1f9.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDIyNDM0NTYsIm5iZiI6MTcwMjI0MzE1NiwicGF0aCI6Ii8xNTIwOTc5OTUvMjg5MzU3NjU0LTY1YWFhNDAxLTFmODEtNDM1ZC05MThjLWQ2MWIyMGZmZjFmOS5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMxMjEwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTIxMFQyMTE5MTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05ZTY4ZDgyYWViMGY4M2U2NmUwMGUxZjI1ODgwODc4MzMwZjJiMmNhZjEyMzFlZjBlMjE3Y2M5NzJjNThkMWE2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.vSzUlW3XeE7gX1qzBk8U57xWYTXqv2r6qEEf74zAax8)

Somehow Fluidd accepted this address on the test, however that is not in the correct format!

The correct format should be `<host>:<moonraker_port>`.

In your case the host is the IP `192.168.0.113` and the Moonraker port is `7126`, so please try this instead:

```text
192.168.0.113:7126
```

**Reply by HereticDezign** [reply]
This : 

192.168.0.113:7125
192.168.0.113:7126

Takes me to both moonraker instances,  how to i get to fluidd from there then ?

**Reply by pedrolamas** [reply, marked as answer]
Those are the addresses you need to add IN Fluidd, not the address to navigate to!

Just use those addresses on the Add Printer dialog inside Fluidd and click save.

**Reply by HereticDezign** [reply]
ok that works, it might be an idea to have the documentation reflect this, as its not very clear on this detail.

thank you for the assistance
